terraform/modules/alb-listener-ip-restriction-rules/main.tf

```diff
@@ -21,22 +8,19 @@ resource "aws_lb_listener_rule" "allow_unrestricted_cidrs" {
 
   condition {
     path_pattern {
-      values = [local.rules[count.index][0]]
+      values = var.restricted_path_patterns
     }
   }
 
   condition {
     source_ip {
-      values = [local.rules[count.index][1]]
+      values = var.fully_trusted_source_ips
     }
   }
 }
 
 resource "aws_lb_listener_rule" "restrict_all_other_ips" {
-  count = length(var.restricted_path_patterns)
-
   listener_arn = var.aws_lb_listener_arn
-  priority     = length(local.rules) + 2 + count.index
```

Comment:
I didn't realise priority is optional on ALB listener rules!

But leaving out an explicit priority on rules might lead to surprising behaviour. From [the docs](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lb_listener_rule):

> Priority (Optional) ... Leaving it unset will automatically set the rule with next available priority after currently existing highest rule. A listener can't have multiple rules with the same priority.

The priority is important, since as there are two rules, 1) requests to a restricted path from a fully trusted source IP are forwarded on; 2) requests to a restricted path from everyone are forbidden. If we reverse the order of these two rules, requests from trusted IPs will be blocked.

Reply:
yes, I think I kept the order correct by making the `denial` rule depends on (install after) the `allow` rule.